GeneratorInterface/Core/interface/TaggedProtonHepMCFilter.h

```diff
@@ -0,0 +1,24 @@
+#ifndef __TAGGEDPROTONHEPMCFILTER__
+#define __TAGGEDPROTONHEPMCFILTER__
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "GeneratorInterface/Core/interface/BaseHepMCFilter.h"
+
+class TaggedProtonHepMCFilter : public BaseHepMCFilter {
+private:
+  const int proton_PDGID_ = 2212;
+  const int neutron_PDGID_ = 2112;
+  const float OneOverbeamEnergy_ = 1.0 / 6500.0;
```

Comment:
could you please modify the energy to float, to make it fit run3 case

Reply:
Hi @menglu21 

Thanks for the suggestions. I introduced a new parameter (center of mass energy) and set it to the run3 case. For run2 simulations, we have different Pythia fragments where we can define the beam energy correspondingly.